src/ui-manager.js

```diff
@@ -41,6 +41,14 @@ class UIManager {
   _translations: {[langKey: string]: Object} = {en: en_translations['en']};
   _locale: string = 'en';
   _uiComponents: Array<KPUIComponent>;
+  /**
+   * Add a component dynamically
+   *
+   * @param {KPUIComponent} component - The component to add
+   * @returns {Function} - Removal function
+   * @memberof UIManager
+   */
+  addComponent: (component: KPUIComponent) => Function;
```

Comment:
I would consider two things - not related to each other
1. provide a noop method as a fallback `() => { return () => {} }`. relevant also for line 181
2. expose a public method for the PlayerAreaProvider instead of passing the method to the upper component and use ref to invoke that method

both not mandatory but might be more solid

Reply:
1. done
2. I tried this option. it's doesn't work since the ref points the `PlayerAreaProvider` logger wrapper